crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -17,7 +18,23 @@ use subspace_networking::Config;
 use subspace_rpc_primitives::FarmerProtocolInfo;
 use tracing::{info, trace, warn};
 
-use crate::{ArchivingFrom, DiskFarm, FarmingArgs};
+struct CallOnDrop(Option<Box<dyn FnOnce() + Send + 'static>>);
+
+impl Drop for CallOnDrop {
+    fn drop(&mut self) {
+        let callback = self.0.take().expect("Only removed on drop; qed");
+        callback();
+    }
+}
+
+impl CallOnDrop {
+    fn new<F>(callback: F) -> Self
+    where
+        F: FnOnce() + Send + 'static,
+    {
+        Self(Some(Box::new(callback)))
+    }
+}
```

Comment:
Should we just factor it to some library?

Reply:
I didn't think it was worth it at the time, it is simple enough to copy-paste.